src/guards/book-exists.ts

```diff
@@ -50,7 +49,7 @@ export class BookExistsGuard implements CanActivate {
    */
   hasBookInApi(id: string) {
     return this.googleBooks.retrieveBook(id)
-      .map(book => this.bookActions.loadBook(book))
+      .map(book => new bookActions.LoadBook(book))
       .do(action => this.store.dispatch(action))
       .map(book => !!book)
       .catch(() => {
```

Comment:
If you are returning an observable of `false`, do you need to manually navigate the router?  In the comment above the `canActivate` method, it says if you return an observable of `false`, that it will fall through to the next route candidate.